activerecord/lib/active_record/migration.rb

```diff
@@ -574,15 +574,15 @@ def nearest_delegate # :nodoc:
       end
 
       # Raises <tt>ActiveRecord::PendingMigrationError</tt> error if any migrations are pending.
-      def check_pending!(connection = Base.connection)
+      def check_pending!(connection = self.class.connection)
```

Comment:
Is there any part of the public API that requires this to be a class method? Can we change it to be an instance method universally, and inject it?